### ISSUE TITLE
v6.0.2.preview - Page start/end character indexes are incorrect and influence audio playback etc.

### ISSUE THREAD

**Issue by SpicyMelonCrab** [JujuAdams/scribble#68]
CREATE EVENT

```GML
scribble_init("fonts","fnt_fifaBig",true);
centerX = obj_displayControl.ideal_width*.5; 
middleY = obj_displayControl.ideal_height*.5; 

height = 200; 
width = 400; 
borderSize = 4;


myPortrait = spr_pilotRabbitOld; 
//myString = "Hi guys let's all go to the temple. This is a great thing we're doing and we're special boys. Let's all go to the temple and let's ahve a pizza party. The best pizza party of your lives. I guarantee it!"; 

myString = "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";
myString += "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";
myString += "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";

string_replace_all(myString," "," [delay,100]"); 
myName = "Name"; 

readSpeed = .25; 
//scribble_draw_set_transform(2,2,0); 
scribble_draw_set_box_align(fa_left,fa_top); 
scribble_draw_set_wrap(-1, width - borderSize*3, height - borderSize*3);
scribbleString = NULLVALUE; 

scribbleName = scribble_draw(0,0,myName);

xScaleTween = TweenFire(id,EaseInQuint,0,false,0,30,image_yscale__,0,1); 
TweenFire(id,EaseInQuint,0,false,0,30,image_xscale__,0,1); 
```

STEP EVENT

```GML
if (!TweenExists(xScaleTween) and scribbleString = NULLVALUE){
	scribbleString = scribble_cache(myString);
	scribble_autotype_fade_in(scribbleString, SCRIBBLE_AUTOTYPE_PER_CHARACTER, readSpeed, 0);
	scribble_autotype_function(scribbleString, scribble_autotype_callback);
}
```

DRAW GUI

```GML
			if (scribbleString != NULLVALUE){
				scribble_draw_set_box_align(fa_left,fa_top); 
				scribble_draw(_left + borderSize*2,_top + borderSize*2,scribbleString);
				var _progress = scribble_autotype_get(scribbleString); 
				if _progress >= 1{
					var _buttonString = "Next";  
					var _onLastPage = scribble_page_on_last(scribbleString); 
					if _onLastPage{
						_buttonString = "Done"
					}
					draw_set_font(fnt_fifaMedium); 
					var _buttWidth = string_width(_buttonString); 
					var _buttHeight = string_width("A"); 
					var _buttonResult = scr_drawGUIButtonAndCheckForButtonPress(_buttonString,_right - _buttWidth*.7 - borderSize,_bottom - _buttHeight*1.1 - borderSize*2,image_alpha,true,NULLVALUE,NULLVALUE,1);
					if (_buttonResult[0] == true){ // BUTTON WAS PRESSED
						if _onLastPage{
							instance_destroy(); 
						}else{
							scribble_page_set(scribbleString, scribble_page_get(scribbleString) + 1);
							scribble_autotype_fade_in(scribbleString,SCRIBBLE_AUTOTYPE_PER_CHARACTER,readSpeed,0); 
						}
						scr_playSound(snd_confirm,false,.1); 
					}
					draw_set_font(fnt_fifa); 
				}
			}
```

**Comment by SpicyMelonCrab** [comment]
`
CREATE
scribble_init("fonts","fnt_fifaBig",true);
centerX = obj_displayControl.ideal_width*.5; 
middleY = obj_displayControl.ideal_height*.5; 

height = 200; 
width = 400; 
borderSize = 4;


myPortrait = spr_pilotRabbitOld; 
//myString = "Hi guys let's all go to the temple. This is a great thing we're doing and we're special boys. Let's all go to the temple and let's ahve a pizza party. The best pizza party of your lives. I guarantee it!"; 

myString = "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";
myString += "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";
myString += "[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.[c_yellow]Setting[/c] [shake]up[/shake][wave]text[/wave] [rainbow]box[/rainbow] [c_green]test[/c] system.";

myString = string_replace_all(myString," "," [delay,150]"); 
myName = "Name"; 

readSpeed = .2; 
//scribble_draw_set_transform(2,2,0); 
scribbleString = NULLVALUE; 

scribbleName = scribble_draw(0,0,myName);

xScaleTween = TweenFire(id,EaseInQuint,0,false,0,30,image_yscale__,0,1); 
TweenFire(id,EaseInQuint,0,false,0,30,image_xscale__,0,1); 

skipCharacterList = ds_list_create();
ds_list_add(skipCharacterList,".",",","!","?","'"); 
`

**Comment by SpicyMelonCrab** [comment]
When the above code runs, the text visually appears to stop rendering before it reaches the correct limit at the end of the first page. However, after it stops visually rendering, the audio specified by scribble_autotype_function(scribbleString, scribble_autotype_callback); continues to run.

When I click the next button and scroll to the next page, the characters that I heard being typed appear to already be rendered. And the text continues to write normally from there.

**Comment by SpicyMelonCrab** [comment]
Note that this error occurs whether SCRIBBLE_CREATE_CHARACTER_ARRAY is set to off or on. But that obviously the audio component is only there if it is set to true.

**Comment by JujuAdams** [comment]
Fixed in [6.0.3.pre](https://github.com/JujuAdams/scribble/releases/tag/6.0.3.pre)
